# Advanced Pune Cafe Sales Dataset Analysis

### Objective

Identify the **top-performing-products** in terms of total revenue. Identify seasonal patterns and any significant anomalies or shifts in sales over the year.

In [100]:
import pandas as pd
import numpy as np

In [101]:
df = pd.read_csv("C:\\Users\\mohin\\Downloads\\Advanced_Pune_Cafe_Sales_2000.csv")
df

order_id        date outlet_location  outlet_manager customer_type  \
0     ORD51036  19-10-2025       Hinjewadi      Priya Nair       Regular   
1     ORD94971  09-10-2025         Kothrud     Arjun Mehta       Tourist   
2     ORD82709  07-10-2025     Viman Nagar  Rohit Deshmukh       Tourist   
3     ORD61541  22-10-2025     Viman Nagar  Rohit Deshmukh       Tourist   
4     ORD20674  07-10-2025         Kothrud     Arjun Mehta           New   
...        ...         ...             ...             ...           ...   
1995  ORD88057  03-10-2025     Viman Nagar  Rohit Deshmukh           New   
1996  ORD71910  26-10-2025     Viman Nagar  Rohit Deshmukh       Tourist   
1997  ORD13597  14-10-2025         Kothrud     Arjun Mehta       Tourist   
1998  ORD37000  18-10-2025   Koregaon Park     Amit Sharma           New   
1999  ORD86916  22-10-2025         Kothrud     Arjun Mehta           New   

     item_category            item_name  quantity_sold  price_per_unit  \
0            Snack  Cheese Garlic Bread              4             245   
1         Beverage           Cappuccino              3             134   
2         Beverage            Green Tea              3             243   
3         Beverage           Cappuccino              1             174   
4          Dessert           Cheesecake              4             175   
...            ...                  ...            ...             ...   
1995      Beverage        Hot Chocolate              4             169   
1996       Dessert              Brownie              1             171   
1997         Snack                Pasta              4             162   
1998         Snack         French Fries              3             234   
1999      Beverage        Hot Chocolate              4             224   

      total_bill  ...  time_of_day  temperature_c day_of_week customer_rating  \
0            980  ...      Morning           34.9      Sunday             3.1   
1            402  ...      Evening           25.3    Thursday             3.0   
2            729  ...      Evening           29.0     Tuesday             4.8   
3            174  ...      Evening           25.7   Wednesday             4.8   
4            700  ...    Afternoon           26.8     Tuesday             4.4   
...          ...  ...          ...            ...         ...             ...   
1995         676  ...      Evening           25.8      Friday             4.9   
1996         171  ...        Night           32.5      Sunday             3.8   
1997         648  ...    Afternoon           32.7     Tuesday             3.1   
1998         702  ...      Morning           30.0    Saturday             3.2   
1999         896  ...      Evening           25.2   Wednesday             3.0   

      discount_percent final_amount    special_event  staff_id  day_type  \
0                   15       833.00  Weekend Special      ST11   Weekend   
1                    0       402.00              NaN       ST4   Weekday   
2                    5       692.55  Weekend Special      ST22   Weekday   
3                    0       174.00     Diwali Offer      ST22   Weekday   
4                   15       595.00      Coffee Fest       ST4   Weekday   
...                ...          ...              ...       ...       ...   
1995                15       574.60  Weekend Special      ST14   Weekday   
1996                 0       171.00      Coffee Fest      ST20   Weekend   
1997                 5       615.60              NaN      ST16   Weekday   
1998                10       631.80              NaN      ST10   Weekend   
1999                 5       851.20              NaN      ST11   Weekday   

     cumulative_sales_outlet  
0                     833.00  
1                     402.00  
2                     692.55  
3                     866.55  
4                     997.00  
...                      ...  
1995               188004.30  
1996               188175.30  
1997               175690.50  
1998    

In [102]:
df.head()


order_id        date outlet_location  outlet_manager customer_type  \
0  ORD51036  19-10-2025       Hinjewadi      Priya Nair       Regular   
1  ORD94971  09-10-2025         Kothrud     Arjun Mehta       Tourist   
2  ORD82709  07-10-2025     Viman Nagar  Rohit Deshmukh       Tourist   
3  ORD61541  22-10-2025     Viman Nagar  Rohit Deshmukh       Tourist   
4  ORD20674  07-10-2025         Kothrud     Arjun Mehta           New   

  item_category            item_name  quantity_sold  price_per_unit  \
0         Snack  Cheese Garlic Bread              4             245   
1      Beverage           Cappuccino              3             134   
2      Beverage            Green Tea              3             243   
3      Beverage           Cappuccino              1             174   
4       Dessert           Cheesecake              4             175   

   total_bill  ...  time_of_day  temperature_c day_of_week customer_rating  \
0         980  ...      Morning           34.9      Sunday             3.1   
1         402  ...      Evening           25.3    Thursday             3.0   
2         729  ...      Evening           29.0     Tuesday             4.8   
3         174  ...      Evening           25.7   Wednesday             4.8   
4         700  ...    Afternoon           26.8     Tuesday             4.4   

   discount_percent final_amount    special_event  staff_id  day_type  \
0                15       833.00  Weekend Special      ST11   Weekend   
1                 0       402.00              NaN       ST4   Weekday   
2                 5       692.55  Weekend Special      ST22   Weekday   
3                 0       174.00     Diwali Offer      ST22   Weekday   
4                15       595.00      Coffee Fest       ST4   Weekday   

  cumulative_sales_outlet  
0                  833.00  
1                  402.00  
2                  692.55  
3                  866.55  
4                  997.00  

[5 rows x 23 columns]

In [103]:
df.tail()

order_id        date outlet_location  outlet_manager customer_type  \
1995  ORD88057  03-10-2025     Viman Nagar  Rohit Deshmukh           New   
1996  ORD71910  26-10-2025     Viman Nagar  Rohit Deshmukh       Tourist   
1997  ORD13597  14-10-2025         Kothrud     Arjun Mehta       Tourist   
1998  ORD37000  18-10-2025   Koregaon Park     Amit Sharma           New   
1999  ORD86916  22-10-2025         Kothrud     Arjun Mehta           New   

     item_category      item_name  quantity_sold  price_per_unit  total_bill  \
1995      Beverage  Hot Chocolate              4             169         676   
1996       Dessert        Brownie              1             171         171   
1997         Snack          Pasta              4             162         648   
1998         Snack   French Fries              3             234         702   
1999      Beverage  Hot Chocolate              4             224         896   

      ...  time_of_day  temperature_c day_of_week customer_rating  \
1995  ...      Evening           25.8      Friday             4.9   
1996  ...        Night           32.5      Sunday             3.8   
1997  ...    Afternoon           32.7     Tuesday             3.1   
1998  ...      Morning           30.0    Saturday             3.2   
1999  ...      Evening           25.2   Wednesday             3.0   

      discount_percent final_amount    special_event  staff_id  day_type  \
1995                15        574.6  Weekend Special      ST14   Weekday   
1996                 0        171.0      Coffee Fest      ST20   Weekend   
1997                 5        615.6              NaN      ST16   Weekday   
1998                10        631.8              NaN      ST10   Weekend   
1999                 5        851.2              NaN      ST11   Weekday   

     cumulative_sales_outlet  
1995               188004.30  
1996               188175.30  
1997               175690.50  
1998               190355.85  
1999               176541.70  

[5 rows x 23 columns]

In [104]:
df.drop(['order_id','staff_id'],axis=1,inplace=True)

In [105]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 21 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   date                     2000 non-null   object 
 1   outlet_location          2000 non-null   object 
 2   outlet_manager           2000 non-null   object 
 3   customer_type            2000 non-null   object 
 4   item_category            2000 non-null   object 
 5   item_name                2000 non-null   object 
 6   quantity_sold            2000 non-null   int64  
 7   price_per_unit           2000 non-null   int64  
 8   total_bill               2000 non-null   int64  
 9   cost_price               2000 non-null   float64
 10  profit                   2000 non-null   float64
 11  payment_mode             2000 non-null   object 
 12  time_of_day              2000 non-null   object 
 13  temperature_c            2000 non-null   float64
 14  day_of_week             

In [106]:
df.describe()

quantity_sold  price_per_unit   total_bill   cost_price       profit  \
count    2000.000000     2000.000000  2000.000000  2000.000000  2000.000000   
mean        2.499500      200.353000   500.813500   350.866920   149.946580   
std         1.113831       58.527866   274.319218   195.203022    88.624357   
min         1.000000      100.000000   100.000000    63.150000    21.620000   
25%         2.000000      150.000000   267.750000   183.082500    79.637500   
50%         3.000000      201.000000   460.000000   316.375000   131.355000   
75%         3.000000      249.000000   696.750000   478.690000   203.172500   
max         4.000000      300.000000  1200.000000   946.730000   477.600000   

       temperature_c  customer_rating  discount_percent  final_amount  \
count    2000.000000      2000.000000       2000.000000   2000.000000   
mean       29.518550         3.993100          7.642500    462.413100   
std         3.165595         0.581938          5.519957    255.380474   
min        24.000000         3.000000          0.000000     85.850000   
25%        26.775000         3.500000          5.000000    246.575000   
50%        29.600000         4.000000         10.000000    421.900000   
75%        32.300000         4.500000         15.000000    636.300000   
max        35.000000         5.000000         15.000000   1200.000000   

       cumulative_sales_outlet  
count              2000.000000  
mean              92518.355200  
std               53543.128595  
min                 203.300000  
25%               46386.687500  
50%               91790.750000  
75%              138889.637500  
max              190768.200000

In [107]:
df.describe(include='all')

date outlet_location outlet_manager customer_type item_category  \
count         2000            2000           2000          2000          2000   
unique          31               5              5             3             3   
top     19-10-2025   Koregaon Park    Amit Sharma       Tourist      Beverage   
freq            65             428            428           675           696   
mean           NaN             NaN            NaN           NaN           NaN   
std            NaN             NaN            NaN           NaN           NaN   
min            NaN             NaN            NaN           NaN           NaN   
25%            NaN             NaN            NaN           NaN           NaN   
50%            NaN             NaN            NaN           NaN           NaN   
75%            NaN             NaN            NaN           NaN           NaN   
max            NaN             NaN            NaN           NaN           NaN   

        item_name  quantity_sold  price_per_unit   total_bill   cost_price  \
count        2000    2000.000000     2000.000000  2000.000000  2000.000000   
unique         17            NaN             NaN          NaN          NaN   
top     Ice Cream            NaN             NaN          NaN          NaN   
freq          145            NaN             NaN          NaN          NaN   
mean          NaN       2.499500      200.353000   500.813500   350.866920   
std           NaN       1.113831       58.527866   274.319218   195.203022   
min           NaN       1.000000      100.000000   100.000000    63.150000   
25%           NaN       2.000000      150.000000   267.750000   183.082500   
50%           NaN       3.000000      201.000000   460.000000   316.375000   
75%           NaN       3.000000      249.000000   696.750000   478.690000   
max           NaN       4.000000      300.000000  1200.000000   946.730000   

        ...  payment_mode time_of_day temperature_c  day_of_week  \
count   ...          2000        2000   2000.000000         2000   
unique  ...             3           4           NaN            7   
top     ...          Card     Morning           NaN    Wednesday   
freq    ...           683         524           NaN          324   
mean    ...           NaN         NaN     29.518550          NaN   
std     ...           NaN         NaN      3.165595          NaN   
min     ...           NaN         NaN     24.000000          NaN   
25%     ...           NaN         NaN     26.775000          NaN   
50%     ...           NaN         NaN     29.600000          NaN   
75%     ...           NaN         NaN     32.300000          NaN   
max     ...           NaN         NaN     35.000000          NaN   

       customer_rating  discount_percent  final_amount    special_event  \
count      2000.000000       2000.000000   2000.000000              593   
unique             NaN               NaN           NaN                3   
top                NaN               NaN           NaN  Weekend Special   
freq               NaN               NaN           NaN              207   
mean          3.993100          7.642500    462.413100              NaN   
std           0.581938          5.519957    255.380474              NaN   
min           3.000000          0.000000     85.850000              NaN   
25%           3.500000          5.000000    246.575000              NaN   
50%           4.000000         10.000000    421.900000              NaN   
75%           4.500000         15.000000    636.300000              NaN   
max           5.000000         15.000000   1200.000000              NaN   

       day_type cumulative_sales_outlet  
count      2000             2000.000000  
unique        2                     NaN  
top     Weekday                     NaN  
freq       1483                     NaN  
mean        NaN            92518.355200  
std         NaN            53543.128595  
min         NaN              203.300000  
25%         NaN            46386.687500  
50%     

In [108]:
df.duplicated().sum()

0

In [109]:
df.isnull().sum()

date                          0
outlet_location               0
outlet_manager                0
customer_type                 0
item_category                 0
item_name                     0
quantity_sold                 0
price_per_unit                0
total_bill                    0
cost_price                    0
profit                        0
payment_mode                  0
time_of_day                   0
temperature_c                 0
day_of_week                   0
customer_rating               0
discount_percent              0
final_amount                  0
special_event              1407
day_type                      0
cumulative_sales_outlet       0
dtype: int64

In [110]:
df["date"] = pd.to_datetime(df["date"])

C:\Users\mohin\AppData\Local\Temp\ipykernel_10416\3228555721.py:1: UserWarning: Parsing dates in %d-%m-%Y format when dayfirst=False (the default) was specified. Pass `dayfirst=True` or specify a format to silence this warning.
  df["date"] = pd.to_datetime(df["date"])


In [111]:
df['Day'] = df['date'].dt.day
df['Month'] = df['date'].dt.month
df['Year'] = df['date'].dt.year

In [112]:
df.drop(['special_event'],axis=1,inplace=True)

In [113]:
df.isnull().sum()

date                       0
outlet_location            0
outlet_manager             0
customer_type              0
item_category              0
item_name                  0
quantity_sold              0
price_per_unit             0
total_bill                 0
cost_price                 0
profit                     0
payment_mode               0
time_of_day                0
temperature_c              0
day_of_week                0
customer_rating            0
discount_percent           0
final_amount               0
day_type                   0
cumulative_sales_outlet    0
Day                        0
Month                      0
Year                       0
dtype: int64

In [114]:
df.drop(['date'],axis=1,inplace=True)

In [115]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 22 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   outlet_location          2000 non-null   object 
 1   outlet_manager           2000 non-null   object 
 2   customer_type            2000 non-null   object 
 3   item_category            2000 non-null   object 
 4   item_name                2000 non-null   object 
 5   quantity_sold            2000 non-null   int64  
 6   price_per_unit           2000 non-null   int64  
 7   total_bill               2000 non-null   int64  
 8   cost_price               2000 non-null   float64
 9   profit                   2000 non-null   float64
 10  payment_mode             2000 non-null   object 
 11  time_of_day              2000 non-null   object 
 12  temperature_c            2000 non-null   float64
 13  day_of_week              2000 non-null   object 
 14  customer_rating         

In [116]:
df.head()

outlet_location  outlet_manager customer_type item_category  \
0       Hinjewadi      Priya Nair       Regular         Snack   
1         Kothrud     Arjun Mehta       Tourist      Beverage   
2     Viman Nagar  Rohit Deshmukh       Tourist      Beverage   
3     Viman Nagar  Rohit Deshmukh       Tourist      Beverage   
4         Kothrud     Arjun Mehta           New       Dessert   

             item_name  quantity_sold  price_per_unit  total_bill  cost_price  \
0  Cheese Garlic Bread              4             245         980      771.03   
1           Cappuccino              3             134         402      267.03   
2            Green Tea              3             243         729      519.16   
3           Cappuccino              1             174         174      105.44   
4           Cheesecake              4             175         700      559.49   

   profit  ... temperature_c day_of_week  customer_rating discount_percent  \
0  208.97  ...          34.9      Sunday              3.1               15   
1  134.97  ...          25.3    Thursday              3.0                0   
2  209.84  ...          29.0     Tuesday              4.8                5   
3   68.56  ...          25.7   Wednesday              4.8                0   
4  140.51  ...          26.8     Tuesday              4.4               15   

   final_amount  day_type  cumulative_sales_outlet Day  Month  Year  
0        833.00   Weekend                   833.00  19     10  2025  
1        402.00   Weekday                   402.00   9     10  2025  
2        692.55   Weekday                   692.55   7     10  2025  
3        174.00   Weekday                   866.55  22     10  2025  
4        595.00   Weekday                   997.00   7     10  2025  

[5 rows x 22 columns]

In [117]:
df.shape

(2000, 22)

In [118]:
df.nunique()

outlet_location               5
outlet_manager                5
customer_type                 3
item_category                 3
item_name                    17
quantity_sold                 4
price_per_unit              201
total_bill                  581
cost_price                 1969
profit                     1935
payment_mode                  3
time_of_day                   4
temperature_c               111
day_of_week                   7
customer_rating              21
discount_percent              4
final_amount               1258
day_type                      2
cumulative_sales_outlet    1999
Day                          31
Month                         1
Year                          1
dtype: int64

In [119]:
df['outlet_location'].value_counts()

outlet_location
Koregaon Park    428
Hinjewadi        405
Viman Nagar      399
Baner            389
Kothrud          379
Name: count, dtype: int64

In [120]:
df[df['outlet_location']=='Hinjewadi']

outlet_location outlet_manager customer_type item_category  \
0          Hinjewadi     Priya Nair       Regular         Snack   
12         Hinjewadi     Priya Nair           New       Dessert   
13         Hinjewadi     Priya Nair       Tourist       Dessert   
18         Hinjewadi     Priya Nair       Regular       Dessert   
25         Hinjewadi     Priya Nair           New       Dessert   
...              ...            ...           ...           ...   
1967       Hinjewadi     Priya Nair       Regular       Dessert   
1974       Hinjewadi     Priya Nair       Tourist      Beverage   
1977       Hinjewadi     Priya Nair       Regular      Beverage   
1986       Hinjewadi     Priya Nair       Regular       Dessert   
1987       Hinjewadi     Priya Nair       Regular         Snack   

                item_name  quantity_sold  price_per_unit  total_bill  \
0     Cheese Garlic Bread              4             245         980   
12                  Donut              1             289         289   
13              Ice Cream              3             293         879   
18             Cheesecake              1             288         288   
25                  Donut              4             200         800   
...                   ...            ...             ...         ...   
1967              Brownie              4             103         412   
1974            Green Tea              3             147         441   
1977        Hot Chocolate              3             141         423   
1986     Chocolate Muffin              4             249         996   
1987                Pasta              1             153         153   

      cost_price  profit  ... temperature_c day_of_week  customer_rating  \
0         771.03  208.97  ...          34.9      Sunday              3.1   
12        183.21  105.79  ...          27.8      Monday              4.9   
13        655.87  223.13  ...          33.9    Saturday              4.8   
18        195.27   92.73  ...          35.0    Thursday              4.8   
25        483.75  316.25  ...          31.1      Monday              4.5   
...          ...     ...  ...           ...         ...              ...   
1967      295.07  116.93  ...          30.6      Friday              3.8   
1974      317.89  123.11  ...          33.7      Friday              3.2   
1977      300.03  122.97  ...          31.7     Tuesday              4.9   
1986      724.14  271.86  ...          32.1      Sunday              3.5   
1987      102.55   50.45  ...          30.4      Monday              4.5   

     discount_percent  final_amount  day_type  cumulative_sales_outlet Day  \
0                  15        833.00   Weekend                   833.00  19   
12                 15        245.65   Weekday                  1078.65  20   
13                 15        747.15   Weekend                  1825.80  25   
18                 10        259.20   Weekday                  2085.00  23   
25                 15        680.00   Weekday                  2765.00  20   
...               ...           ...       ...                      ...  ..   
1967                5        391.40   Weekday                188920.95   3   
1974                5        418.95   Weekday                189339.90  17   
1977                5        401.85   Weekday                189741.75  14   
1986               10        896.40   Weekend                190638.15  19   
1987               15        130.05   Weekday                190768.20   6   

      Month  Year  
0        10  2025  
12       10  2025  
13       10  2025  
18       10  2025  
25       10  2025  
...     ...   ...  
1967     10  2025  
1974     10  2025  
1977     10  2025  
1986     10  2025  
1987     10  2025  

[405 rows x 22 columns]

In [121]:
df['item_category'].value_counts()

item_category
Beverage    696
Snack       665
Dessert     639
Name: count, dtype: int64

In [122]:
df[(df['item_category'] == 'Beverage') & (df['outlet_location'] == 'Baner')]

outlet_location outlet_manager customer_type item_category    item_name  \
7              Baner    Sneha Patil           New      Beverage   Cappuccino   
19             Baner    Sneha Patil       Tourist      Beverage    Green Tea   
24             Baner    Sneha Patil       Regular      Beverage   Cappuccino   
27             Baner    Sneha Patil       Tourist      Beverage  Cold Coffee   
30             Baner    Sneha Patil           New      Beverage  Masala Chai   
...              ...            ...           ...           ...          ...   
1887           Baner    Sneha Patil       Tourist      Beverage  Masala Chai   
1933           Baner    Sneha Patil       Tourist      Beverage    Green Tea   
1935           Baner    Sneha Patil       Regular      Beverage  Masala Chai   
1971           Baner    Sneha Patil       Tourist      Beverage  Masala Chai   
1981           Baner    Sneha Patil       Tourist      Beverage    Green Tea   

      quantity_sold  price_per_unit  total_bill  cost_price  profit  ...  \
7                 3             243         729      479.32  249.68  ...   
19                3             213         639      457.25  181.75  ...   
24                2             160         320      241.22   78.78  ...   
27                1             164         164      106.94   57.06  ...   
30                4             111         444      295.78  148.22  ...   
...             ...             ...         ...         ...     ...  ...   
1887              3             151         453      334.03  118.97  ...   
1933              1             252         252      177.46   74.54  ...   
1935              2             197         394      292.07  101.93  ...   
1971              2             273         546      394.69  151.31  ...   
1981              4             137         548      434.97  113.03  ...   

     temperature_c day_of_week  customer_rating discount_percent  \
7             29.2     Tuesday              4.7               10   
19            28.2   Wednesday              4.0                5   
24            33.5      Monday              3.3               15   
27            29.1   Wednesday              4.8               15   
30            34.7   Wednesday              4.1                5   
...            ...         ...              ...              ...   
1887          29.6      Sunday              4.1               15   
1933          31.9      Sunday              3.9                0   
1935          32.0    Thursday              3.4                5   
1971          28.7      Friday              3.4                5   
1981          28.6     Tuesday              4.1               10   

      final_amount  day_type  cumulative_sales_outlet Day  Month  Year  
7           656.10   Weekday                   656.10   7     10  2025  
19          607.05   Weekday                  1425.15   8     10  2025  
24          272.00   Weekday                  1879.05  27     10  2025  
27          139.40   Weekday                  2018.45  29     10  2025  
30          421.80   Weekday                  2440.25  15     10  2025  
...            ...       ...                      ...  ..    ...   ...  
1887        385.05   Weekend                174450.55  19     10  2025  
1933        252.00   Weekend                175390.80  12     10  2025  
1935        374.30   Weekday                175765.10  23     10  2025  
1971        518.70   Weekday                177094.70  24     10  2025  
1981        493.20   Weekday                177871.00  21     10  2025  

[135 rows x 22 columns]

In [123]:
df['item_name'].value_counts()

item_name
Ice Cream              145
Iced Latte             145
Brownie                140
Cappuccino             131
Hot Chocolate          129
Paneer Wrap            124
Chocolate Muffin       122
French Fries           120
Cold Coffee            120
Donut                  117
Cheesecake             115
Burger                 114
Cheese Garlic Bread    110
Pasta                   99
Veg Sandwich            98
Masala Chai             87
Green Tea               84
Name: count, dtype: int64

In [124]:
df['payment_mode'].value_counts()

payment_mode
Card    683
UPI     660
Cash    657
Name: count, dtype: int64

In [125]:
df['time_of_day'].value_counts()

time_of_day
Morning      524
Afternoon    495
Night        492
Evening      489
Name: count, dtype: int64

In [126]:
item_with_customer_rating = df.groupby('item_name')['customer_rating'].mean().sort_values(ascending=False)
item_with_customer_rating

item_name
Masala Chai            4.079310
Cheese Garlic Bread    4.079091
Burger                 4.073684
Cheesecake             4.070435
Cappuccino             4.064122
Hot Chocolate          4.022481
Veg Sandwich           4.012245
Paneer Wrap            3.986290
Green Tea              3.985714
Iced Latte             3.964828
Chocolate Muffin       3.963115
Ice Cream              3.951724
French Fries           3.951667
Pasta                  3.946465
Brownie                3.945714
Donut                  3.940171
Cold Coffee            3.890000
Name: customer_rating, dtype: float64

- Masala chai, cheese garlic bread, and burger have the highest average customer ratings around 4.07-4.08 out of 5 and the lower average item is cold coffee with rating of 3.89.

In [128]:
df['customer_rating'].max()

5.0

In [129]:
df[df['customer_rating'] == 5.0]

outlet_location  outlet_manager customer_type item_category  \
11     Koregaon Park     Amit Sharma           New      Beverage   
21           Kothrud     Arjun Mehta       Tourist      Beverage   
104      Viman Nagar  Rohit Deshmukh           New      Beverage   
109        Hinjewadi      Priya Nair           New       Dessert   
110            Baner     Sneha Patil       Regular       Dessert   
169          Kothrud     Arjun Mehta       Regular         Snack   
174          Kothrud     Arjun Mehta           New      Beverage   
229    Koregaon Park     Amit Sharma       Tourist       Dessert   
272            Baner     Sneha Patil       Tourist      Beverage   
274          Kothrud     Arjun Mehta       Regular       Dessert   
334    Koregaon Park     Amit Sharma       Tourist      Beverage   
393    Koregaon Park     Amit Sharma           New      Beverage   
467          Kothrud     Arjun Mehta       Regular         Snack   
481          Kothrud     Arjun Mehta           New       Dessert   
581          Kothrud     Arjun Mehta           New      Beverage   
671      Viman Nagar  Rohit Deshmukh       Tourist      Beverage   
876    Koregaon Park     Amit Sharma       Tourist       Dessert   
926    Koregaon Park     Amit Sharma       Regular      Beverage   
972            Baner     Sneha Patil           New       Dessert   
983          Kothrud     Arjun Mehta       Tourist       Dessert   
1057       Hinjewadi      Priya Nair           New       Dessert   
1151       Hinjewadi      Priya Nair           New      Beverage   
1176     Viman Nagar  Rohit Deshmukh       Tourist       Dessert   
1195     Viman Nagar  Rohit Deshmukh           New         Snack   
1211   Koregaon Park     Amit Sharma       Tourist      Beverage   
1249       Hinjewadi      Priya Nair       Tourist         Snack   
1301   Koregaon Park     Amit Sharma       Regular         Snack   
1307   Koregaon Park     Amit Sharma       Tourist      Beverage   
1309         Kothrud     Arjun Mehta       Tourist         Snack   
1373     Viman Nagar  Rohit Deshmukh           New       Dessert   
1429         Kothrud     Arjun Mehta       Tourist      Beverage   
1457       Hinjewadi      Priya Nair           New         Snack   
1518   Koregaon Park     Amit Sharma           New       Dessert   
1519           Baner     Sneha Patil       Tourist         Snack   
1610       Hinjewadi      Priya Nair       Tourist      Beverage   
1611     Viman Nagar  Rohit Deshmukh       Tourist         Snack   
1623   Koregaon Park     Amit Sharma       Regular      Beverage   
1684       Hinjewadi      Priya Nair           New         Snack   
1711     Viman Nagar  Rohit Deshmukh       Tourist      Beverage   
1715           Baner     Sneha Patil       Regular      Beverage   
1741         Kothrud     Arjun Mehta           New       Dessert   
1782     Viman Nagar  Rohit Deshmukh       Tourist      Beverage   
1802   Koregaon Park     Amit Sharma       Tourist       Dessert   
1837       Hinjewadi      Priya Nair       Regular       Dessert   
1884       Hinjewadi      Priya Nair       Regular      Beverage   
1957   Koregaon Park     Amit Sharma           New      Beverage   

                item_name  quantity_sold  price_per_unit  total_bill  \
11            Masala Chai              1             100         100   
21          Hot Chocolate              3             206         618   
104            Cappuccino              1             240         240   
109            Cheesecake              1             111         111   
110      Chocolate Muffin              2             254         508   
169                Burger              1             228         228   
174            Iced Latte              1             245         245   
229            Cheesecake              4             173         692   
272             Green Tea              1             126         126   
274            Cheesecake              3             243         729   
334           Co

In [130]:
df[(df['customer_rating'] == 5.0) & (df['time_of_day']=='Morning')]

outlet_location  outlet_manager customer_type item_category  \
110            Baner     Sneha Patil       Regular       Dessert   
581          Kothrud     Arjun Mehta           New      Beverage   
1211   Koregaon Park     Amit Sharma       Tourist      Beverage   
1429         Kothrud     Arjun Mehta       Tourist      Beverage   
1518   Koregaon Park     Amit Sharma           New       Dessert   
1611     Viman Nagar  Rohit Deshmukh       Tourist         Snack   
1711     Viman Nagar  Rohit Deshmukh       Tourist      Beverage   
1741         Kothrud     Arjun Mehta           New       Dessert   
1782     Viman Nagar  Rohit Deshmukh       Tourist      Beverage   
1802   Koregaon Park     Amit Sharma       Tourist       Dessert   
1837       Hinjewadi      Priya Nair       Regular       Dessert   

                item_name  quantity_sold  price_per_unit  total_bill  \
110      Chocolate Muffin              2             254         508   
581            Iced Latte              1             189         189   
1211           Iced Latte              4             226         904   
1429           Iced Latte              3             203         609   
1518     Chocolate Muffin              2             276         552   
1611  Cheese Garlic Bread              2             209         418   
1711            Green Tea              3             278         834   
1741                Donut              1             255         255   
1782        Hot Chocolate              4             235         940   
1802                Donut              4             185         740   
1837              Brownie              4             287        1148   

      cost_price  profit  ... temperature_c day_of_week  customer_rating  \
110       339.91  168.09  ...          25.8     Tuesday              5.0   
581       134.96   54.04  ...          33.4      Monday              5.0   
1211      577.18  326.82  ...          32.8    Saturday              5.0   
1429      449.13  159.87  ...          25.0     Tuesday              5.0   
1518      414.01  137.99  ...          33.5      Monday              5.0   
1611      275.51  142.49  ...          25.6      Sunday              5.0   
1711      558.60  275.40  ...          32.2      Friday              5.0   
1741      170.85   84.15  ...          26.5     Tuesday              5.0   
1782      629.53  310.47  ...          28.7      Friday              5.0   
1802      486.99  253.01  ...          28.1   Wednesday              5.0   
1837      848.10  299.90  ...          34.0      Sunday              5.0   

     discount_percent  final_amount  day_type  cumulative_sales_outlet Day  \
110                 0        508.00   Weekday                 11739.15  21   
581                10        170.10   Weekday                 49845.85  20   
1211                0        904.00   Weekend                117573.35   4   
1429                5        578.55   Weekday                121956.35  21   
1518               10        496.80   Weekday                142017.90  13   
1611                0        418.00   Weekend                139622.35   5   
1711                0        834.00   Weekday                153798.85   3   
1741                0        255.00   Weekday                148715.55  28   
1782               10        846.00   Weekday                164681.20  10   
1802                0        740.00   Weekday                170168.80  29   
1837                0       1148.00   Weekend                174161.05  12   

      Month  Year  
110      10  2025  
581      10  2025  
1211     10  2025  
1429     10  2025  
1518     10  2025  
1611     10  2025  
1711     10  2025  
1741     10  2025  
1782     10  2025  
1802     10  2025  
1837     10  2025  

[11 rows x 22 columns]

In [131]:
df['quantity_sold'].max()

4

In [132]:
item_sales = df.groupby('item_name')['quantity_sold'].sum().reset_index()
item_with_highest_quantity = item_sales.loc[item_sales['quantity_sold'].idxmax()]

In [133]:
print(item_with_highest_quantity)

item_name        Ice Cream
quantity_sold          358
Name: 11, dtype: object


- 'Ice cream' is the item with the highest quantity sold (358 units).

In [135]:
df['total_bill'].max()

1200

In [136]:
item_category_with_highest_total_bill = df.groupby('item_category')['total_bill'].sum().idxmax()
item_category_with_highest_total_bill

'Beverage'

In [137]:
most_profitable_item = df.loc[df['profit'].idxmax()]

In [138]:
most_profitable_item

outlet_location            Koregaon Park
outlet_manager               Amit Sharma
customer_type                        New
item_category                   Beverage
item_name                  Hot Chocolate
quantity_sold                          4
price_per_unit                       300
total_bill                          1200
cost_price                         722.4
profit                             477.6
payment_mode                        Card
time_of_day                        Night
temperature_c                       27.3
day_of_week                       Friday
customer_rating                      3.6
discount_percent                      10
final_amount                      1080.0
day_type                         Weekday
cumulative_sales_outlet         128234.4
Day                                   10
Month                                 10
Year                                2025
Name: 1326, dtype: object

In [139]:
least_profitable_item = df.loc[df['profit'].idxmin()]

In [140]:
least_profitable_item

outlet_location                Kothrud
outlet_manager             Arjun Mehta
customer_type                      New
item_category                 Beverage
item_name                   Cappuccino
quantity_sold                        1
price_per_unit                     102
total_bill                         102
cost_price                       80.38
profit                           21.62
payment_mode                      Cash
time_of_day                    Morning
temperature_c                     28.3
day_of_week                     Monday
customer_rating                    4.5
discount_percent                     0
final_amount                     102.0
day_type                       Weekday
cumulative_sales_outlet        39506.4
Day                                  6
Month                               10
Year                              2025
Name: 466, dtype: object

In [141]:
df_daily_sales = df.groupby(['day_of_week', 'item_category'])['quantity_sold'].sum().reset_index().sort_values(by='day_of_week' , ascending=True)

In [142]:
df_daily_sales

day_of_week item_category  quantity_sold
0       Friday      Beverage            270
1       Friday       Dessert            240
2       Friday         Snack            293
3       Monday      Beverage            226
4       Monday       Dessert            204
5       Monday         Snack            207
6     Saturday      Beverage            218
7     Saturday       Dessert            218
8     Saturday         Snack            221
11      Sunday         Snack            246
10      Sunday       Dessert            188
9       Sunday      Beverage            224
12    Thursday      Beverage            292
13    Thursday       Dessert            225
14    Thursday         Snack            264
15     Tuesday      Beverage            234
16     Tuesday       Dessert            201
17     Tuesday         Snack            225
18   Wednesday      Beverage            262
19   Wednesday       Dessert            278
20   Wednesday         Snack            263

In [191]:
sales_by_day_type = df.groupby('day_type').size()
sales_by_day_type

day_type
Weekday    1483
Weekend     517
dtype: int64

In [195]:
most_sold_day_type = sales_by_day_type.sort_values(ascending=False).head(1)
most_sold_day_type

day_type
Weekday    1483
dtype: int64

In [199]:
most_popular_day = sales_by_day_type.idxmax()
most_popular_day

'Weekday'

In [203]:
total_sales_on_day = sales_by_day_type.max()
total_sales_on_day

1483

In [211]:
print(f'The Most Popular Day Type is: {most_popular_day} and (Total Sales: {total_sales_on_day})')


The Most Popular Day Type is: Weekday and (Total Sales: 1483)


In [223]:
df_filtered = df[df['day_type'] == most_popular_day]
item_sales_on_day = df_filtered.groupby('item_name').size()
most_sold_item = item_sales_on_day.idxmax()
item_sales_count = item_sales_on_day.max()
most_sold_item

'Hot Chocolate'

In [225]:
item_sales_count

107

In [229]:
print(f'The item sold most frequently on {most_popular_day} is: {most_sold_item}')
print(f"  (It sold {item_sales_count} times on {most_popular_day}s)")

# Optional: Show all items on that day type
print("\nTop Items on that Day:")
print(item_sales_on_day.sort_values(ascending=False))

The item sold most frequently on Weekday is: Hot Chocolate
  (It sold 107 times on Weekdays)

Top Items on that Day:
item_name
Hot Chocolate          107
Brownie                101
Chocolate Muffin       101
Cappuccino              98
Iced Latte              94
Ice Cream               94
French Fries            94
Donut                   93
Paneer Wrap             90
Cold Coffee             87
Cheesecake              82
Veg Sandwich            80
Burger                  77
Cheese Garlic Bread     77
Pasta                   75
Green Tea               67
Masala Chai             66
dtype: int64


In [235]:
# count of occurrence of each payment mode
payment_counts = df['payment_mode'].value_counts()
payment_counts

payment_mode
Card    683
UPI     660
Cash    657
Name: count, dtype: int64

In [241]:
# Group by Location and Item, and sum the Revenue ---
# We use 'final_amount' as the revenue column.
revenue_by_item_location = df.groupby(['outlet_location', 'item_name'])['final_amount'].sum().reset_index()

# Rename the column for clarity
revenue_by_item_location = revenue_by_item_location.rename(columns={'final_amount': 'Total_Revenue'})

# Create a clean, readable pivot table ---
# Convert the result to a pivot table where locations are columns
pivot_table = revenue_by_item_location.pivot_table(
    index='item_name', 
    columns='outlet_location', 
    values='Total_Revenue', 
    fill_value=0 # Fill missing values (items not sold in a location) with 0
)

# Sort the columns (locations) alphabetically for a cleaner look
pivot_table = pivot_table.sort_index(axis=1)

# Display the top 10 rows and the full data summary
print("--- Pivot Table: Total Revenue (INR) of Items by Location ---")
print(pivot_table)



--- Pivot Table: Total Revenue (INR) of Items by Location ---
outlet_location         Baner  Hinjewadi  Koregaon Park   Kothrud  Viman Nagar
item_name                                                                     
Brownie              13626.55   13263.75       10217.10  12149.65     10472.15
Burger                9567.05    7888.15        9355.45  12744.25     10581.20
Cappuccino           12931.20   14915.70       11936.00   9147.45     14048.45
Cheese Garlic Bread  11160.40   10468.35       10933.55   8951.05      7776.35
Cheesecake           10644.65    6249.25       14055.45  10120.00      9148.25
Chocolate Muffin      8626.90   13794.35       10006.15   8679.95     17744.90
Cold Coffee           8723.30   14466.50        9111.80  12307.30     10551.20
Donut                 9405.80   11159.20       13939.05  10881.95      8242.75
French Fries         10228.80   10858.40       12866.15  12069.80     11635.70
Green Tea             7459.05    8624.30        7967.00   5582.65    